### PR TITLE
Prevent product-preview from zeroing data when stale

### LINKED ISSUE #3513: Product update after timeout during preview zeros fields
Using zc157, current alpha (note that the issue's also present in zc156, possibly earlier).

In the admin (site's set up with the demo products), navigate to edit a product (I chose "Matrox G200 MMS", pID=1).  Click the "Preview" button and wait until the timeout jQuery script indicates that the session's timed out.

Click the "Log in again" button, log in and you're taken back to the product's "Preview".  Clicking "Update" at this time wipes various fields within the product's definition.

Here's the phpMyAdmin view of that product's original definition:
image

... and here's that view after clicking that "Update" button:
image

The 'update' action has set the `products_type`, `products_price`, `products_quantity`, `products_status` (and others) to 0 and has wiped the product's model number to an empty string.

### PR DESCRIPTION
**Pull request by drbyte**
Fixes #3513 : "Product update after timeout during preview zeros fields"

/cc @lat9

### CODE REVIEW

File: admin/product.php [hunk 1]
```diff
@@ -35,12 +35,25 @@
       }
       break;
     case 'new_product_preview':
+      if (!isset($_POST['master_categories_id'])
+          || ((isset($_POST['products_model']) ? $_POST['products_model'] : '') . (isset($_POST['products_url']) ? implode('', $_POST['products_url']) : '') . (isset($_POST['products_name']) ? implode('', $_POST['products_name']) : '') . (isset($_POST['products_description']) ? implode('', $_POST['products_description']) : '') != '')
```
Comment (drbyte) [inline admin/product.php:39]: Maybe these checks are excessive? (this 2nd line of concatenated var comes from existing code)

File: admin/product.php [hunk 2]
```diff
@@ -35,12 +35,25 @@
...
+//          zen_redirect(zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'cPath=' . $cPath . (isset($_GET['pID']) ? '&pID=' . $_GET['pID'] : '')));
+      }
       if (file_exists(DIR_WS_MODULES . $zc_products->get_handler($product_type) . '/new_product_preview.php')) {
         require(DIR_WS_MODULES . $zc_products->get_handler($product_type) . '/new_product_preview.php');
       } else {
         require(DIR_WS_MODULES . 'new_product_preview.php');
       }
       break;
+    case 'new_product_preview_meta_tags':
+      if (!isset($_POST['products_price_sorter']) || !isset($_POST['products_model'])) {
```
Comment (drbyte) [inline admin/product.php:52]: These POST vars were taken from the collect_info_metatags form.

**Comment by zcwilt** [comment]
$_POST['master_categories_id'] isn't set on the first call to new_products_preview

$_POST['product_type'] might be a better test as this is guaranteed to be set to something.

**Comment by scottcwilson** [comment]
hang on i have a fix

**Pull request merged**